core/src/main/kotlin/ch/ergon/dope/resolvable/clause/model/OrderByClause.kt

```diff
@@ -2,49 +2,58 @@ package ch.ergon.dope.resolvable.clause.model
 import ch.ergon.dope.resolvable.formatToQueryStringWithSymbol
-import ch.ergon.dope.validtype.StringType
 import ch.ergon.dope.validtype.ValidType
 
-enum class OrderByType(val type: String) {
+enum class OrderByType(val queryString: String) {
     ASC("ASC"),
     DESC("DESC"),
 }
 
 private const val ORDER_BY = "ORDER BY"
 
-open class SelectOrderByClause<T : ValidType>(private val stringField: Field<StringType>, private val parentClause: ISelectGroupByClause<T>) :
-    ISelectOrderByClause<T> {
-
+class OrderExpression(private val typeExpression: TypeExpression<out ValidType>, private val orderByType: OrderByType? = null) : Resolvable {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
-        val parentDopeQuery = parentClause.toDopeQuery(manager)
-        val stringDopeQuery = stringField.toDopeQuery(manager)
+        val typeDopeQuery = typeExpression.toDopeQuery(manager)
         return DopeQuery(
-            queryString = formatToQueryStringWithSymbol(parentDopeQuery.queryString, ORDER_BY, stringDopeQuery.queryString),
-            parameters = parentDopeQuery.parameters.merge(stringDopeQuery.parameters),
+            queryString = "${typeDopeQuery.queryString}${orderByType?.queryString?.let { " $it" } ?: ""}",
+            parameters = typeDopeQuery.parameters,
         )
     }
 }
 
-class SelectOrderByTypeClause<T : ValidType>(
-    private val stringField: Field<StringType>,
-    private val orderByType: OrderByType,
+class SelectOrderByClause<T : ValidType>(
+    private val orderExpression: OrderExpression,
+    private vararg val additionalOrderExpressions: OrderExpression,
     private val parentClause: ISelectGroupByClause<T>,
-) : SelectOrderByClause<T>(stringField, parentClause) {
-
+) : ISelectOrderByClause<T> {
     override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
         val parentDopeQuery = parentClause.toDopeQuery(manager)
-        val stringDopeQuery = stringField.toDopeQuery(manager)
+        val orderExpressionDopeQuery = orderExpression.toDopeQuery(manager)
+        val additionalOrderExpressions = additionalOrderExpressions.map { it.toDopeQuery(manager) }
         return DopeQuery(
             queryString = formatToQueryStringWithSymbol(
                 parentDopeQuery.queryString,
                 ORDER_BY,
-                stringDopeQuery.queryString + " $orderByType",
+                orderExpressionDopeQuery.queryString,
+                *additionalOrderExpressions.map { it.queryString }.toTypedArray(),
+            ),
+            parameters = parentDopeQuery.parameters.merge(
+                orderExpressionDopeQuery.parameters,
+                *additionalOrderExpressions.map { it.parameters }.toTypedArray(),
             ),
-            parameters = parentDopeQuery.parameters.merge(stringDopeQuery.parameters),
         )
     }
+
+    fun thenOrderBy(typeExpression: TypeExpression<out ValidType>, orderByType: OrderByType? = null) =
```

Comment:
not as extension function? How do we do it in similar cases?

Reply:
I have it like that because the user can than see the function dirrectly without needing to import it.